docs/firefighting/index.md

```diff
@@ -205,9 +205,583 @@ It may happen, specially if the client crashes or has kernel oops, that a connec
+
+```
+root@redis0:/opt/data/redis# cd /opt/data/redis/
+root@redis0:/opt/data/redis# redis-cli
+127.0.0.1:6379> CONFIG SET maxmemory 4gb
+OK
+(1.06s)
+root@redis0:/opt/data/redis# rm temp-rewriteaof-\*.aof
+```
+
+# Riak CS
+
+## Pinging a Riak Machine
+
+A simple way to check if riak is working is to use the `check_services` management command.  If that's showing it as down, make sure the services are running on all machines.  Riak usually runs on port 8080.  You should be able to curl any riak machine like so and see that you get a response from the Riak CS server.
+
+```
+$ curl -I hqriak0.internal-va.commcarehq.org:8080
+HTTP/1.1 405 Method Not Allowed
+Server: Riak CS
+Date: Mon, 21 Nov 2016 23:24:00 GMT
+Content-Type: application/xml
+Content-Length: 216
+Allow: GET
+```
+
+Another thing you can do to get a visual on riak is log on to a riak machine, `su` to root, and run `riak-admin cluster status`
+
+```
+root@staging-hqdjango2:/home/cchq/www/staging/current# riak-admin cluster status
+---- Cluster Status ----
+Ring ready: true
+
++----------------------------------------------------------------------+------+-------+-----+-------+
+|                                 node                                 |status| avail |ring |pending|
++----------------------------------------------------------------------+------+-------+-----+-------+
+| (C) riak-hqdjango2-staging@hqdjango2-staging.internal.commcarehq.org |valid |  up   | 25.0|  --   |
+|     riak-hqdjango3-staging@hqdjango3-staging.internal.commcarehq.org |valid |  up   | 25.0|  --   |
+|     riak-hqdjango4-staging@hqdjango4-staging.internal.commcarehq.org |valid |  up   | 25.0|  --   |
+|     riak-hqdjango5-staging@hqdjango5-staging.internal.commcarehq.org |valid |  up   | 25.0|  --   |
++----------------------------------------------------------------------+------+-------+-----+-------+
+
+Key: (C) = Claimant; availability marked with '!' is unexpected
+```
+
+## Riak is Out of Memory
+
+The most common problem we have had with Riak and Riak CS is running [out of memory](https://docs.basho.com/riak/kv/2.1.4/using/repair-recovery/failure-recovery/#out-of-memory). Most of the time this is non-actionable; nodes will crash and automatically restart and operation usually returns to normal without intervention. If a node is consistently running out of memory then it is likely necessary to either allocate more memory (usually done by creating a new node with more memory) or by allocating more swap space (can affect performance).
+
+##What do do when one or more Riak nodes has a corrupt database
```

Comment:
needs space between `##` and `What`

Reply:
https://github.com/dimagi/commcare-cloud/pull/2612/commits/8ea0e728fb497c29ae9ddb09ae37b41bdf8e0771